tests/tokenizer-tests.ts

```diff
@@ -169,6 +169,26 @@ QUnit.test('A simple tag', function(assert) {
   assert.deepEqual(tokens, [startTag('div')]);
 });
 
+QUnit.test('A simple tag with leading non alpha chars', function(assert) {
+  let tokens = tokenize('<_div>');
+  assert.deepEqual(tokens, [startTag('_div')]);
+
+  tokens = tokenize('<$div>');
+  assert.deepEqual(tokens, [startTag('$div')]);
```

Comment:
let's add emoji case!

```html
<😀> Smile! </😀>
```

Reply:
i was tempted to add it :)
but
<img width="468" alt="Screenshot 2025-06-03 at 22 06 45" src="https://github.com/user-attachments/assets/1f83b494-b45a-496b-bf61-3240110fd3d0" />

Reply:
we could still add it: 
1. Assert that it throw error if emoji is used in the beginning of tag name
2. Assert ok if it's in the middle

<img width="842" alt="image" src="https://github.com/user-attachments/assets/5dda431c-4053-4ad6-a834-33f390342673" />

Reply:
Yeah this distinction is important for parsing Javascript identifiers, the leading character is more restricted than the rest of them. 

The code here should only deal with things that are valid leading characters, because it's controlling when we enter the tag states. We leave those states based only on space and `/>`, so it's necessarily quite tolerant.

Reply:
at that point where I introduced the change. It already is in the state of `tagOpen`. because it detected the `<`. 
after that it tries to find the beginning of the tag name.
if we just skip characters here, it will just hide the issue.
I think the tokenizer should just allow anything at this stage and let other tools decide if its a valid name.

given that a Javascript variable can not be created with an invalid name, it will be impossible to reference on from a tagname.

so i mean it would be possible to write `<😀div>` and it would not matter because we cannot write that name in JS.
either way. currently the tokenizer would just skip the `😀` and return `div` as tag name

Reply:
So, in other words. the tokenizer should not be opiniated about it. just tokenize the thing :)

Reply:
<img width="1275" alt="image" src="https://github.com/user-attachments/assets/f929050c-5a2c-4e78-a780-01842b3656ca" />
Seems very strange

Reply:
Is that with the current fix? Before it would just skip any non alpha characters

Reply:
But we should also make sure tfat this works with any char in the middle